torch/csrc/autograd/python_engine.cpp

```diff
@@ -130,10 +134,27 @@ void compute_partial_exec_callbacks(const function_list& roots,
   }
 }
 
+static bool _reinitialize_engine = false;
+
+static void _maybe_reinitialize_engine_after_fork() {
+  // This is "probably" thread-safe because the flag is set in a fork handler
```

Comment:
I think it's safer to make it `std::atomic<bool>` with a static assert that it's lock-free